github-runner-manager/src/github_runner_manager/manager/models.py

```diff
@@ -179,3 +179,16 @@ def as_dict(self) -> dict[str, str]:
             metadata as a dict.
         """
         return {k: v for k, v in asdict(self).items() if v is not None}
+
+
+@dataclass
+class RunnerContext:
+    """Information provided by the platform provider needed to spawn a runner.
+
+    Attributes:
+        shell_run_script: Script to run the platform agent.
+        ingress_tcp_ports_to_open: Ports to be opened in the cloud provider.
+    """
+
+    shell_run_script: str
+    ingress_tcp_ports_to_open: list[int] = field(default_factory=lambda: [])
```

Comment:
Is `_to_open` a necessary part of the variable name?
I feel it might be implied and makes the variable name a bit long.
I am fine either way.